// clyde.js/miscellaneous.js
module.exports = function(string){
const { Prefix, Token } = require('./index.js')
const discord = require('discord.js')
const client = new discord.Client();
// When The Bot Is Ready
console.log(`| Miscellaneous Commands : Ready |`);
//Get Stock Info
client.on('message', message => {
  const prefix = Prefix || ""
if (message.content === Prefix + 'ping') {
const exampleEmbed = { // Make It Stylish By Using Embeds
	color: "RANDOM", // Color : Could Be Hex Code Or Anything
	title: `Pong ðŸ“`, 
};

message.channel.send({ embed: exampleEmbed }); 


}
});


client.on('message', message => {
  const prefix = Prefix || ""
if(message.content === Prefix + "help"){
const exampleEmbed2 = { 
	author: {
		name: `Hey ${message.author.username}`,
		icon_url: message.author.avatarURL({ dynamic: true, format: 'png', size: 1024 }),
	},
	color: "RANDOM", 
	description: `[**Commands**](https://discord.com)\n\n**Finance Related**\n\n[${Prefix}get](https://discord.com) - Get Stock Quote\n[${Prefix}getfinancial](https://discord.com) - Get Financial Datas\n[${Prefix}getcompany](https://discord.com) - Get Company Datas\n[${Prefix}getnspa](https://discord.com) - Get netSharePurchaseActivity\n\n**Miscellaneous**\n\n[${Prefix}ping](https://discord.com) - Pong\n[${Prefix}help] - Get Help`, 
	timestamp: new Date(), 
};
message.channel.send({ embed: exampleEmbed2 }); 



}
});

client.on('message', message => {
   if(message.content == Prefix+"uptime"){
     let seconds = Math.floor(message.client.uptime / 1000);
    let minutes = Math.floor(seconds / 60);
    let hours = Math.floor(minutes / 60);
    let days = Math.floor(hours / 24);

    seconds %= 60;
    minutes %= 60;
    hours %= 24;

const exampleEmbed1 = {
	color: "RANDOM",
	title: `Uptime`,
	author: {
		name: `${message.author.username}`,
		icon_url: message.author.avatarURL({ dynamic: true, format: 'png', size: 1024 }),
	},

description: `**Days** : ${days}\n**Hours** : ${hours}\n**Minutes** : ${minutes}\n **Seconds** : ${seconds}`,

	timestamp: new Date(),
	footer: {
		icon_url: client.user.avatarURL({ dynamic: true, format: 'png', size: 1024 }) ,
	},
};
message.channel.send({ embed: exampleEmbed1 })

   }



})



client.login(Token)

}
